spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/FeignClientFactoryBean.java

```diff
@@ -275,4 +273,99 @@ public boolean isSingleton() {
 		return true;
 	}
 
+	public Class<?> getType() {
+		return type;
+	}
+
+	public void setType(Class<?> type) {
+		this.type = type;
+	}
+
+	public String getName() {
+		return name;
+	}
+
+	public void setName(String name) {
+		this.name = name;
+	}
+
+	public String getUrl() {
+		return url;
+	}
+
+	public void setUrl(String url) {
+		this.url = url;
+	}
+
+	public String getPath() {
+		return path;
+	}
+
+	public void setPath(String path) {
+		this.path = path;
+	}
+
+	public boolean isDecode404() {
+		return decode404;
+	}
+
+	public void setDecode404(boolean decode404) {
+		this.decode404 = decode404;
+	}
+
+	public ApplicationContext getApplicationContext() {
+		return applicationContext;
+	}
+
+	public Class<?> getFallback() {
+		return fallback;
+	}
+
+	public void setFallback(Class<?> fallback) {
+		this.fallback = fallback;
+	}
+
+	public Class<?> getFallbackFactory() {
+		return fallbackFactory;
+	}
+
+	public void setFallbackFactory(Class<?> fallbackFactory) {
+		this.fallbackFactory = fallbackFactory;
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) return true;
+		if (o == null || getClass() != o.getClass()) return false;
+		FeignClientFactoryBean that = (FeignClientFactoryBean) o;
+		return Objects.equals(applicationContext, that.applicationContext) &&
+				decode404 == that.decode404 &&
+				Objects.equals(fallback, that.fallback) &&
+				Objects.equals(fallbackFactory, that.fallbackFactory) &&
+				Objects.equals(name, that.name) &&
+				Objects.equals(path, that.path) &&
+				Objects.equals(type, that.type) &&
+				Objects.equals(url, that.url);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(applicationContext, decode404, fallback, fallbackFactory,
+				name, path, type, url);
+	}
+
+	@Override
+	public String toString() {
+		return new StringBuilder("FeignClientFactoryBean{")
```

Comment:
I learned that Spring has a [ToStringCreator](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/core/style/ToStringCreator.html) that we should enforce in the future.